pkg/syncer/controller/syncerconfigmanager.go

```diff
@@ -129,6 +132,99 @@ func (s *SyncerConfigManager) upsertNamespaceScoped(syncerConfig edgev1alpha1.Sy
 	s.syncConfigManager.upsert(edgeSyncConfig)
 }
 
+type typeFlatNamespacedObject struct {
+	APIVersion string
+	v1.GroupResource
+	Namespace string
+	Names     []string
+}
+
+type typeTableOfFlatNamespacedObject struct {
+	FlatNamespacedObjects []typeFlatNamespacedObject
+}
+
+func (t *typeTableOfFlatNamespacedObject) filter(predicate func(row typeFlatNamespacedObject) bool) *typeTableOfFlatNamespacedObject {
+	flatNamespacedObjects := []typeFlatNamespacedObject{}
+	for _, obj := range t.FlatNamespacedObjects {
+		if predicate(obj) {
+			flatNamespacedObjects = append(flatNamespacedObjects, obj)
+		}
+	}
+	return &typeTableOfFlatNamespacedObject{FlatNamespacedObjects: flatNamespacedObjects}
+}
+
+func (t *typeTableOfFlatNamespacedObject) getAllNamespaces() []string {
+	namespaces := sets.String{}
+	for _, obj := range t.FlatNamespacedObjects {
+		namespaces.Insert(obj.Namespace)
+	}
+	return namespaces.List()
+}
+
+func (s *SyncerConfigManager) upsertNamespacedObjects(syncerConfig edgev1alpha1.SyncerConfig, upstreamGroupResourcesList []*restmapper.APIGroupResources) {
+	flatNamespacedObjects := []typeFlatNamespacedObject{}
+	for _, nsObject := range syncerConfig.Spec.NamespacedObjects {
+		for _, obj := range nsObject.ObjectsByNamespace {
+			flatNamespacedObjects = append(flatNamespacedObjects, typeFlatNamespacedObject{
+				APIVersion:    nsObject.APIVersion,
+				GroupResource: nsObject.GroupResource,
+				Namespace:     obj.Namespace,
+				Names:         obj.Names,
+			})
+		}
+	}
+	namespacedObjectsTable := typeTableOfFlatNamespacedObject{FlatNamespacedObjects: flatNamespacedObjects}
+
+	requiredNamespaces := namespacedObjectsTable.filter(func(row typeFlatNamespacedObject) bool { return len(row.Names) > 0 }).getAllNamespaces()
+
+	s.logger.V(3).Info("upsert namespaced objects as syncerConfig to syncConfigManager stores", "syncerConfigName", syncerConfig.Name, "numNamespaces", len(requiredNamespaces))
+	if lgr := s.logger.V(4); lgr.Enabled() {
+		for _, agrs := range upstreamGroupResourcesList {
+			lgr.Info("APIGroupResources", "group", agrs.Group, "bar", agrs.VersionedResources)
```

Comment:
Is there a bit of unfinished editing here?

Reply:
I think no.

Reply:
Why the key "bar"?

Reply:
"bar" would come from https://github.com/yana1205/kubestellar/blob/syncer-support-namespacedobject/pkg/syncer/controller/syncerconfigmanager.go#L94... I just copied and pasted it. It's probably typo of "ver". I will fix it.